core/src/apps/bitcoin/sign_tx/decred.py

```diff
@@ -208,6 +232,54 @@ def write_tx_output(
             writers.write_uint16(w, DECRED_SCRIPT_VERSION)
         writers.write_bytes_prefixed(w, script_pubkey)
 
+    def validate_sstx_submission(self, txo: TxOutput) -> bytearray:
+        assert txo.address is not None
+        return scripts_decred.output_script_sstxsubmissionpkh(txo.address)
+
+    def validate_sstx_commitment_owned(self, txo: TxOutput) -> bytearray:
+        assert txo.op_return_data is not None  # checked in sanitize_tx_output
+        # Verify that the address this script pays to is owned by the wallet.
+        wantpkh, wantamt = scripts_decred.pkh_from_sstxcommitment(txo.op_return_data)
+        if wantamt != self.approver.total_in:
+            raise wire.DataError("sstxcommitment has incorrect amount")
+        key = self.keychain.derive(txo.address_n)
+        gotpkh = ecdsa_hash_pubkey(key.public_key(), self.coin)
+        if gotpkh != wantpkh:
+            raise wire.DataError("sstxcommitment does not pay to this wallet")
+        return scripts_decred.output_script_paytoopreturn(txo.op_return_data)
+
+    def validate_sstx_change(self, txo: TxOutput) -> bytearray:
+        assert txo.address is not None  # checked in sanitize_tx_output
```

Comment:
we could instead check that `txo.address` is _the_ particular address -- ISTM the address that decodes to null pubkey is always the same, right?

it would be nicer if we enforced that the address is `""`, but the sanitizer function will not let that through, and that is not worth changing for this particular usecase.

Another option, however, is not requiring the host to send the change output at all, and instead generate it on the Trezor side.

Reply:
> we could instead check that `txo.address` is _the_ particular address -- ISTM the address that decodes to null pubkey is always the same, right?

I think the address will be different for Mainnet, Testnet, Simnet and Regnet because of the different prefixes and checksums.

> it would be nicer if we enforced that the address is `""`, but the sanitizer function will not let that through, and that is not worth changing for this particular usecase.

I tend to agree.

> Another option, however, is not requiring the host to send the change output at all, and instead generate it on the Trezor side.

That seems unsystematic.

I'd just leave as is.

Reply:
OK

Reply:
no, actually let's flip the logic and, _if_ we want to check the address validity, generate a zero-pubkey address on Trezor and compare that to the provided one

I'm also OK with simply ignoring the value of the address field

Reply:
How about this 19b632df63904f364a57f46ac9bedff38780ef9f?

Reply:
yes, this looks reasonable. But please update the comment on `output_script_sstxchange`. And perhaps extract the fixed scriptpubkey value to a constant?

Reply:
Done 7d2cb70e8e222ff38e9a1a8d5d282aad08c87334.